src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -506,6 +507,11 @@ private static GpioDriver GetBestDriverForBoardOnWindows()
             return new RaspberryPi3Driver();
         }
 
+        if (baseBoardProduct == RaspberryPi5Product || baseBoardProduct.StartsWith($"{RaspberryPi5Product} "))
+        {
+            return new LibGpiodDriver();
```

Comment:
Shouldn't this be in the Linux method? AFAIK there's no supported windows version for rpi at this time and libgpiod is not for windows.

Reply:
For now, let's keep this simple. As you've already noticed above, Windows isn't really supported any more (the defunct Windows 10 Iot is only supported on the Raspberry Pi 3, and for Windows 11, there are no officially supported versions for any Pi at all at this time). 

So for now, move this code to `GetBestDriverForLinux()` so at least we get a valid driver there. If I understand correctly, you need to do `new LibGpiodDriver(4)` though. After this, I would say this PR is ready to get merged.

After that, investigation is needed onto the next steps. We probably need a fully new `RaspberryPi5Driver` then, that handles the GPIO device for the new RP1 chip. Don't forget to take a look at the `RaspberryPiBoard` class, which contains the logic to program the mux for the different GPIO alternate functions. This logic will likely also change.

Reply:
Ok done... 

(Ignoring the brain farts of commits - We can squash)... It works...

![20231013_124046](https://github.com/dotnet/iot/assets/7139618/2bb2e746-6e74-4a24-919f-1fb3ebd1a36f)

Of course, if you use a Pin Numbering scheme of Board instead of Logical it throws a Platform Not Supported error (I like the board numbering scheme personally).

Reply:
I tried PWM and a Servo out of interest and it doesn't seem to work interestingly... I wonder is this a limitation of the LibGpioDriver?

Reply:
> I tried PWM and a Servo out of interest and it doesn't seem to work interestingly... I wonder is this a limitation of the LibGpioDriver?

What kind of issue is exhibiting? Any specific error?

Reply:
No errors... Just no PWM out of the pin oddly...

Reply:
Quite strange. Could you please check the other pins with a scope to see if the issue is only a mapping problem?

Reply:
Yup... I'll also try python to double check it's not some other fault!

Reply:
Yup... I'll also try python to double check it's not some other fault!

Reply:
I'm assuming you have added the corresponding lines to `boot/config.txt` (I'm expecting that to still be the same).

Reply:
I've enabled pwm in the Raspberry Pi config tool... (Same thing I think)

Reply:
With Python and `gpiozero` I get jittery servo movement from BCM18... But there's movement...

Reply:
> With Python and `gpiozero` I get jittery servo movement from BCM18... But there's movement...

Could you re-try with C# after running the Python code?

Reply:
Nothing sadly... gpiozero, while jittery, works... I sure it's bit bashed PWM in gpiozero by default given that its jitterry and gives this warning;

`/usr/lib/python3/dist-packages/gpiozero/output_devices.py:1509: PWMSoftwareFallback: To reduce servo jitter, use the pigpio pin factory.See https://gpiozero.readthedocs.io/en/stable/api_output.html#servo for more info
  warnings.warn(PWMSoftwareFallback(`

No movement at all for .NET... Which will be trying to use Hardware PWM.

With that in mind... When I try software PWM from the bindings it works... But not ideal...

```
using SoftwarePwmChannel pwmChannel1 = new SoftwarePwmChannel(18, 50, 0.5, true);
            using ServoMotor servoMotor1 = new ServoMotor(
                pwmChannel1,
                180,
                900,
                2100);

```

Reply:
There is probably some new initialization stuff needed to make it work on the RPi5 but it's hard to say.

Reply:
I'll have a dig around and a play.... See what I can turn up...

Reply:
BTW... I tried adding the following to the config.txt file and rebooting;

```
dtoverlay=pwm-2chan,pin=18,func=2,pin2=13,func2=4
```

But, no dice.

Worth trying eh.